### PR TITLE
Use Optimint node instead of Tendermint Node

### PR DESCRIPTION
**Pull request by tzdybal**
This is minimalistic approach to replace Tendermint Node with Optimint.

Probably more options will be required for Optimint, and maybe some
cleaner way of node object instantiation should be added.

Ideally, changes similar to this should be done with Starport.

### CODE REVIEW

**Comment by tzdybal** [comment]
Ref: https://github.com/lazyledger/lazyledger-core/pull/166, https://github.com/lazyledger/optimint/pull/13

**Comment by evan-forbes** [comment]
The lazyledger-app is currently calling the [`startInProcess`](https://github.com/lazyledger/cosmos-sdk/blob/6ba0b2b8354cca73859b8e85c2a94f3739f8cdea/server/start.go#L201) function when it boots. We might want to think about making switching between optimint and tendermint/ll-core configurable.

**Comment by musalbas** [comment]
Yeah, the node should be configurable rather than hard coded in a way that can be merged upstream, as the lazyledger-app uses Tendermint but rollups will use optimint.

File: server/start.go [hunk 1]
```diff
@@ -235,27 +234,29 @@ func startInProcess(ctx *Context, clientCtx client.Context, appCreator types.App
...
+	//	cfg,
+	//	pvFile,
+	//	nodeKey,
+	//	proxy.NewLocalClientCreator(app),
+	//	genDocProvider,
+	//	node.DefaultDBProvider,
+	//	node.DefaultMetricsProvider(cfg.Instrumentation),
+	//	ctx.Logger,
+	//)
+	tmNode, err = optinode.NewNode(proxy.NewLocalClientCreator(app), ctx.Logger)
```
Comment (liamsi) [inline server/start.go:259]: Note that this fork of the SDK was mainly created to use lazyledger-core which will still provide a tendermint-consensus node (and not be run with optimint).

On the long run, I'd prefer that any changes necessary to the sdk will be "replayed" via some scaffolding tool that devs use to create their application. For now we certainly need a way to track those changes though. But we won't merge this changes to master: the LL main-chain will still need consensus and hence nodes will run a lazyledger-core (tendermint) node.

**Comment by liamsi** [comment]
Let#s figure out https://github.com/lazyledger/lazyledger-core/pull/166/files#r583498992 first. Would it be possible to extract the interface here and only have changes in the SDK that we could then replay with some scaffolding?

Thanks @tzdybal !

**Comment by tzdybal** [comment]
I had to extract "Local" RPC client implementation from Tendermint, and embedded it into Optimint. Thanks to this, no changes in Tendermint are required. 

Only very small changes (easy to "scaffold") in cosmos-sdk to use Optimint instead of Tendermint.

**Review by liamsi — APPROVED** [review]
Looks good. Start adding (integration / e2e-style) tests.

**Pull request merged**